modules/apps/transfer/ibc_module.go

```diff
@@ -136,9 +142,10 @@ func (IBCModule) OnChanOpenAck(
 	_ string,
 	counterpartyVersion string,
 ) error {
-	if counterpartyVersion != types.Version {
-		return errorsmod.Wrapf(types.ErrInvalidVersion, "invalid counterparty version: expected %s, got %s", types.Version, counterpartyVersion)
+	if !slices.Contains(types.SupportedVersions, counterpartyVersion) {
```

Comment:
As written this seems incorrect. We should be checking technically that the counterparty version is the version we already chose in INIT. Not that it is one of the supported versions

Reply:
But if we change the above to not return an error but instead propose a different version then this would be correct.

https://github.com/cosmos/ibc-go/pull/6175#discussion_r1575068425

My suggestion is to just change the `OnChanOpenTry` logic